db/primary_migrate/20220129181752_create_sign_in_restrictions.rb

```diff
@@ -0,0 +1,12 @@
+class CreateSignInRestrictions < ActiveRecord::Migration[6.1]
+  def change
+    create_table :sign_in_restrictions do |t|
+      t.integer :user_id, null: false
+      t.string :service_provider
+
+      t.timestamps
+    end
+
+    add_index :sign_in_restrictions, [:user_id, :service_provider], name: :index_sign_in_restrictions_on_user_id_and_service_provider
```

Comment:
Do we want unique?

Reply:
Probably. I kinda waivered because it seems like it might make it tricky to ban large batches of users. But presumably we'll follow on with some more tooling which can check for an existing record.